metis_client/helpers.py

```diff
@@ -126,11 +126,11 @@ def http_to_metis_error_map(status: int) -> Type[MetisError]:
 def metis_error_to_raise(dto: MetisErrorDTO):
     "Raise MetisErrorDTO"
     if isinstance(dto["error"], dict):
-        msg = str(dto["error"]["message"])
+        msg = str(dto["error"].get("message", "Unexpected server error"))
     else:
         msg = dto["error"]
-    err = http_to_metis_error_map(dto["status"])
-    raise err(status=dto["status"], message=msg)
+    err = http_to_metis_error_map(dto.get("status"))
+    raise err(status=dto.get("status"), message=msg)
```

Comment:
It's useless. `dto` of type `MetisErrorDTO` always has a `status` field. There is a type guard `is_metis_error_dto()`.

Reply:
Same as above: KeyError occurs under certain erroneous server responses

Reply:
There are two callers: `raise_on_metis_error` and `raise_on_metis_error_in_event`. I think `raise_on_metis_error_in_event` doesn't property check type at runtime.

Reply:
Okay, I'm doing it now. Please, wait.

Reply:
Oof, patch is too big. Here is a commit: https://github.com/metis-science/metis-client/pull/38/commits/6a28c0696b2acecb993104ac051cffcca8348673 Revert it if you don't like it.

I think the `KeyError` arises because of data from the event stream. This is now very strictly checked. Type guards and wrappers are also tested.